sapi/cli/php_cli_server.c

```diff
@@ -1639,17 +1642,32 @@ static int php_cli_server_client_read_request_on_path(php_http_parser *parser, c
 static int php_cli_server_client_read_request_on_query_string(php_http_parser *parser, const char *at, size_t length)
 {
 	php_cli_server_client *client = parser->data;
-	client->request.query_string = pestrndup(at, length, 1);
-	client->request.query_string_len = length;
+	if (client->request.query_string == NULL) {
+		client->request.query_string = pestrndup(at, length, 1);
+		client->request.query_string_len = length;
+	} else {
+		client->request.query_string = perealloc(client->request.query_string, client->request.query_string_len + length + 1, 1);
```

Comment:
Seems rather unlikely to matter, but we can play it safe and use safe_perealloc here.

Reply:
I've added the assertions now; do you still think it's better to use safe_perealloc()?

Reply:
No, I think it's fine.